### PR TITLE
7 setup login messagebar and signin states

### PR DESCRIPTION
**Pull request by DurtDiv3r**
Create MessageBar for Login status. Add UseCase for saveSignIn state.

### CODE REVIEW

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.islaharper.dawnofandroid.R
+import com.islaharper.dawnofandroid.domain.model.MessageBarState
+import kotlinx.coroutines.delay
+import java.net.ConnectException
+import java.net.SocketTimeoutException
+
+@Composable
+fun MessageBar(messageBarState: MessageBarState) {
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt:41]: Is `MessageBarState` a suitable parameter for a Composable? Is it stable?

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        )
+    }
+}
+
+@Composable
+fun Message(
+    messageBarState: MessageBarState,
+    errorMessage: String = "",
+) {
+    val messageBarColour = if (messageBarState.error != null) {
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt]: This composable has a lot of switching depending on whether it has an error state or not. Is there any possibility of decomposing the problem in a different way to create simpler Composables, even if it ends up with a bit of duplication?

Alternatively, you can make some of the logic here (like for instance the message bar content color, and the icon) into parameters for the Composable. It doesn't matter if the Composable has a lot of params, especially if you add sensible defaults. Doing it this way means the complexity is pulled down into the ViewModel or at least a Composable that is higher up in the tree.

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            },
+            color = messageBarContentColour,
+            style = MaterialTheme.typography.labelLarge,
+            overflow = TextOverflow.Ellipsis,
+            maxLines = 1,
+        )
+    }
+}
+
+@Preview(name = "Light")
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt:119]: Thank you for doing these, they are useful previews.

File: app/src/main/java/com/islaharper/dawnofandroid/ui/theme/Theme.kt [hunk 4]
```diff
@@ -83,7 +83,7 @@ private val DarkColors = darkColorScheme(
 fun DawnOfAndroidTheme(
     darkTheme: Boolean = isSystemInDarkTheme(),
     // Dynamic color is available on Android 12+
-    dynamicColor: Boolean = true,
+    dynamicColor: Boolean = false,
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/ui/theme/Theme.kt:86]: Why are we turning off dynamic color?

Comment (DurtDiv3r) [reply app/src/main/java/com/islaharper/dawnofandroid/ui/theme/Theme.kt:86]: I'm adding option to toggle light/dark mode and another for dynamic on/off. Was setting it to off as the default

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt (hunk 2 above)
Comment (DurtDiv3r) [reply app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt]: Committed a change with some default parameters. I think I'm on the same page as what you were suggesting...

Comment (drawers) [reply app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt]: Thanks, the commit is definitely an improvement 🎉 
Can we go through this together in our meeting on Monday? I will extend the meeting to an hour

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt (hunk 1 above)
Comment (DurtDiv3r) [reply app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt:41]: Refactored MessageBarState to a sealed class. Refactored MessageBar.kt.

**Review by drawers — APPROVED** [review]

**Pull request merged**